contracts/PositionSpotFactory.sol

```diff
@@ -0,0 +1,179 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+
+import "./interfaces/ISpotFactory.sol";
+import "./libraries/types/SpotFactoryStorage.sol";
+import "./libraries/helper/Errors.sol";
+
+import "@positionex/matching-engine/contracts/interfaces/IMatchingEngineAMM.sol";
+
+contract PositionSpotFactory is
+    ISpotFactory,
+    SpotFactoryStorage,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable
+{
+    address public templatePair;
+
+    //address public owner;
+
+    //    modifier onlyOwner() {
+    //        require(owner == msg.sender, Errors.VL_ONLY_OWNER);
+    //        _;
+    //    }
+
+    constructor() {}
+
+    function createPairManager(
+        address quoteAsset,
+        address baseAsset,
+        uint256 basisPoint,
+        uint256 baseBasisPoint,
+        uint128 maxFindingWordsIndex,
+        uint128 initialPip,
+        uint128 pipRange,
+        uint32 tickSpace
+    ) external nonReentrant {
+        address creator = msg.sender;
+        // ToDo validate pair info
+
```

Comment:
- check require `quoteAsset ` and `baseAsset ` haven't pair used yet.
- check require `basisPoint `, `baseBasisPoint`, `maxFindingWordsIndex `, `initialPip `, `pipRange `, `tickSpace ` > 0